src/mlpack/tests/ann_layer_test.cpp

```diff
@@ -390,6 +390,135 @@ BOOST_AUTO_TEST_CASE(SimpleLinearLayerTest)
+  size_t kernelWidth = 4, kernelHeight = 5;
+  AtrousConvolution<>* module = new AtrousConvolution<>(inSize, outSize,
+      kernelWidth, kernelHeight, 6, 7, std::make_tuple(8, 9),
+      std::make_tuple(10, 11), 12, 13, 14, 15);
+  module->Reset();
+  RandomInitialization().Initialize(module->Weight(), kernelWidth, kernelHeight,
+      inSize * outSize);
+  module->Bias().ones();
+
+  BOOST_CHECK_EQUAL_COLLECTIONS(module->Weight().begin(),
+      module->Weight().end(), module->Parameters().begin(),
+      module->Parameters().end() - outSize);
+
+  BOOST_CHECK_EQUAL_COLLECTIONS(module->Bias().begin(),
+      module->Bias().end(), module->Parameters().end() - outSize,
+      module->Parameters().end());
+
+  BOOST_REQUIRE_EQUAL(module->Weight().n_rows, kernelWidth);
+  BOOST_REQUIRE_EQUAL(module->Weight().n_cols, kernelHeight);
+  BOOST_REQUIRE_EQUAL(module->Weight().n_slices, inSize * outSize);
+  BOOST_REQUIRE_EQUAL(module->Bias().n_rows, outSize);
+  BOOST_REQUIRE_EQUAL(module->Bias().n_cols, 1);
+  BOOST_REQUIRE_EQUAL(module->Parameters().n_rows,
+      (outSize * inSize * kernelWidth * kernelHeight) + outSize);
+
+  delete module;
+}
+
+/**
+ * Convolution module weight initialization test.
+ */
+BOOST_AUTO_TEST_CASE(ConvolutionLayerWeightInitializationTest)
+{
+  size_t inSize = 2, outSize = 3;
+  size_t kernelWidth = 4, kernelHeight = 5;
+  Convolution<>* module = new Convolution<>(inSize, outSize,
+      kernelWidth, kernelHeight, 6, 7, std::tuple<size_t, size_t>(8, 9),
+      std::tuple<size_t, size_t>(10, 11), 12, 13, "none");
+  module->Reset();
+  RandomInitialization().Initialize(module->Weight(), kernelWidth, kernelHeight,
+      inSize * outSize);
+  module->Bias().ones();
+
+  BOOST_CHECK_EQUAL_COLLECTIONS(module->Weight().begin(),
+      module->Weight().end(), module->Parameters().begin(),
+      module->Parameters().end() - outSize);
+
+  BOOST_CHECK_EQUAL_COLLECTIONS(module->Bias().begin(),
+      module->Bias().end(), module->Parameters().end() - outSize,
+      module->Parameters().end());
+
+  BOOST_REQUIRE_EQUAL(module->Weight().n_rows, kernelWidth);
+  BOOST_REQUIRE_EQUAL(module->Weight().n_cols, kernelHeight);
+  BOOST_REQUIRE_EQUAL(module->Weight().n_slices, inSize * outSize);
+  BOOST_REQUIRE_EQUAL(module->Bias().n_rows, outSize);
+  BOOST_REQUIRE_EQUAL(module->Bias().n_cols, 1);
+  BOOST_REQUIRE_EQUAL(module->Parameters().n_rows,
+      (outSize * inSize * kernelWidth * kernelHeight) + outSize);
+
+  delete module;
+}
+
+/**
+ * Transposed Convolution module weight initialization test.
+ */
+BOOST_AUTO_TEST_CASE(TransposedConvolutionLayerWeightInitializationTest)
+{
+  size_t inSize = 1, outSize = 1;
+  size_t kernelWidth = 4, kernelHeight = 4;
+  TransposedConvolution<>* module = new TransposedConvolution<>(inSize, outSize,
+      kernelWidth, kernelHeight, 1, 1, 1, 1, 5, 5, 6, 6);
+  module->Reset();
+  RandomInitialization().Initialize(module->Weight(), kernelWidth, kernelHeight,
+      inSize * outSize);
```

Comment:
That's weird. The test should fail too.
```suggestion
  module->Weight().randu();
```
Perhaps it didn't fail since you didn't change the values of `inSize`, `outSize`, `kernelWidth`, `kernelHeight` (see the comment below).

Reply:
Maybe because the `weight` here is of size `4x4x1` (kind of matrix).